cmd/global-heal.go

```diff
@@ -173,6 +153,28 @@ func (er *erasureObjects) healErasureSet(ctx context.Context, buckets []string,
 		}
 	}
 
+	info, err := tracker.disk.DiskInfo(ctx, false)
+	if err != nil {
+		return fmt.Errorf("unable to get disk information before healing it: %w", err)
+	}
+
+	var numHealers uint64
+
+	if numCores := uint64(runtime.GOMAXPROCS(0)); info.NRRequests > numCores {
+		numHealers = numCores / 4
+	} else {
+		numHealers = info.NRRequests / 4
+	}
+	if numHealers == 0 || numHealers == 1 {
+		numHealers = 4
+	}
+	// allow overriding this value as well..
+	if v := globalHealConfig.GetWorkers(); v > 0 {
+		numHealers = uint64(v)
+	}
+
+	jt, _ := workers.New(int(numHealers))
+
```

Comment:
let's also update https://github.com/minio/minio/blob/e3fbcaeb729f5dd02f4b7c96c10b952c1d7dfe2a/docs/config/README.md?plain=1#L274

To show this configurable value

Reply:
We have to say that this is Linux only optimization, for example on Windows and other OS we will use `number of cores / 4`